### PR TITLE
[terra-framework-docs] Slider implementation guide and docs update

### PR DESCRIPTION
**Pull request by nikhitasharma**
### Summary

**What was changed:**
Added new Accessibility and Implementation guide for Terra Slider

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [x] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9722 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/Examples.2/CustomSlider.3.doc.mdx [hunk 1]
```diff
@@ -1,3 +1,9 @@
 import CustomSlider from './CustomSlider?dev-site-example';
 
+# Custom Slider
+
+### Description
+
+This example demonstrates a [Slider](/components/cerner-terra-framework-docs/slider/about) with custom field label,  minimum and maximum labels. This example uses 'minimumLabel' and 'maximumLabel' props to set custom labels.
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/slider/Examples.2/CustomSlider.3.doc.mdx:7]:
```suggestion
This example demonstrates a [Slider](/components/cerner-terra-framework-docs/slider/about) with a custom field label and minimum and maximum labels. This example uses 'minimumLabel' and 'maximumLabel' props to set custom labels. These labels are also used to construct the visually hidden text used to assist screen reader users.
```

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx [hunk 2]
```diff
@@ -2,31 +2,91 @@ import { Notice } from "@cerner/terra-docs";
...
+* [Implementation](/components/cerner-terra-framework-docs/slider/about#implementation)
+* [Testing](/components/cerner-terra-framework-docs/slider/about#testing)
+* [Properties](/components/cerner-terra-framework-docs/slider/about#properties)
+* [Terra Standards](/components/cerner-terra-framework-docs/slider/about#terra-standards)
+
+## Overview
+
 Terra Slider provides a way to select a value from a given range by sliding a thumb along a bar to track the value.
 
 <Notice variant="caution" ariaLevel="2">
```
Comment (chrismichalewicz) [inline packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx:24]: We need to stop using the Caution banner for these types of messages as I believe they require Reg Affairs approval. I think updating this to an Important! banner with a green background similar to what Charles did in the Terra Table documentation should work.

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/Examples.2/SliderWithOnChangeCallback.4.doc.mdx [hunk 3]
```diff
@@ -1,3 +1,9 @@
 import SliderWithOnChangeCallback from './SliderWithOnChangeCallback?dev-site-example';
 
+# Slider with onChange callback
+
+### Description
+
+This example demonstrates a [Slider](/components/cerner-terra-framework-docs/slider/about) with an onChange Callback using the 'onChange' callback prop.
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/slider/Examples.2/SliderWithOnChangeCallback.4.doc.mdx:7]:
```suggestion
This example demonstrates a [Slider](/components/cerner-terra-framework-docs/slider/about) with an onChange Callback using the 'onChange' callback prop and setting the 'value' prop.
```

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx [hunk 4]
```diff
@@ -2,31 +2,91 @@ import { Notice } from "@cerner/terra-docs";
...
 Terra Slider is an input component used to quickly select an imprecise value within a range such as a zoom level. It does not yet include features required for entering precise values.
 </Notice>
 
+<DefaultSlider />
+
+### Features
+Supports:
+* Management of custom slider label, minimum label and maximum label.
```
Comment (chrismichalewicz) [inline packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx:32]: nit: Add an 'a' in front of custom
"Management of a custom slider lable, minimum label and maximum label."

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx [hunk 5]
```diff
@@ -2,31 +2,91 @@ import { Notice } from "@cerner/terra-docs";
...
+* Setting default value for slider.
```
Comment (chrismichalewicz) [inline packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx:33]: Nit: Update it to "Setting a default value for the slider"

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx [hunk 6]
```diff
@@ -2,31 +2,91 @@ import { Notice } from "@cerner/terra-docs";
...
+* Disabling of slider.
```
Comment (chrismichalewicz) [inline packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx:34]: Nit: Update it to "Disabling of the slider"

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/Examples.2/SliderWithValueUpdate.5.doc.mdx [hunk 7]
```diff
@@ -1,3 +1,9 @@
 import SliderWithValueUpdate from './SliderWithValueUpdate?dev-site-example';
 
+# Slider with Value Update
+
+### Description
+
+This example demonstrates a [Slider](/components/cerner-terra-framework-docs/slider/about) with value updates using the 'value' prop and 'onChange' prop.
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/slider/Examples.2/SliderWithValueUpdate.5.doc.mdx:7]: I think this and the 'onChange' example could be merged.

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx [hunk 8]
```diff
@@ -2,31 +2,91 @@ import { Notice } from "@cerner/terra-docs";
...
+* Externally updating the slider value using the 'value' prop.
 
-## Getting Started
+Refer to the [examples](/components/cerner-terra-framework-docs/slider/about#examples) section for detailed demonstration of the features supported by Slider.
+
+## Examples
+|Link to Example| Description|
+|-|-|
+|[BasicSlider](/components/cerner-terra-framework-docs/slider/examples/default-slider)|An example of basic slider using the required props 'label', 'minimumValue', 'maximumValue', 'value'.|
+|[CustomSlider](/components/cerner-terra-framework-docs/slider/examples/custom-slider)|An example demonstrating a slider with a custom labels using the 'minimumLabel' and 'maximumLabel' props.|
```
Comment (chrismichalewicz) [inline packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx:44]: Remove the "a" before "custom"

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx [hunk 9]
```diff
@@ -2,31 +2,91 @@ import { Notice } from "@cerner/terra-docs";
...
+Supports:
+* Management of custom slider label, minimum label and maximum label.
+* Setting default value for slider.
+* Disabling of slider.
+* Making other updates based on slider value change using the 'onChange' callback prop.
+* Externally updating the slider value using the 'value' prop.
 
-## Getting Started
+Refer to the [examples](/components/cerner-terra-framework-docs/slider/about#examples) section for detailed demonstration of the features supported by Slider.
+
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx:39]: An Important block here explaining:

Consumers are given the ability to programmatically change the value of the slider without the user directly interacting with it. In most cases this isn't recommended. However, if this is done, ensure the UI makes it _very clear_ that the value was changed and that screen readers announce the change.

**Review by chrismichalewicz — CHANGES_REQUESTED** [review]

**Review by eawww — COMMENTED** [review]
Nice Job!

Just a few things.

**Comment by mjpalazzo** [comment]
@nikhitasharma - great job on the documentation. I have one minor observation - in the example section of the about page BasicSlider is listed, but in the vertical navigation menu this item is called Default Slider.

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx (hunk 2 above)
Comment (nikhitasharma) [reply packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx]: Addressed all comments [here](https://github.com/cerner/terra-framework/pull/1945/commits/715f9208bff9a79fe767e7cc2d97decac2dff644 )

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/Examples.2/CustomSlider.3.doc.mdx (hunk 1 above)
Comment (nikhitasharma) [reply packages/terra-framework-docs/src/terra-dev-site/doc/slider/Examples.2/CustomSlider.3.doc.mdx]: Addressed all comments [here](https://github.com/cerner/terra-framework/pull/1945/commits/715f9208bff9a79fe767e7cc2d97decac2dff644)

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/Examples.2/SliderWithValueUpdate.5.doc.mdx (hunk 7 above)
Comment (nikhitasharma) [reply packages/terra-framework-docs/src/terra-dev-site/doc/slider/Examples.2/SliderWithValueUpdate.5.doc.mdx]: @eawww I remember there was request to have an additional example with programmatic value updates from dev review on my actual Slider component implementation PR. I also think both there examples show different ways of code implementation for slider. 
I have changed the wording on the programmatically setting value example to specifically try and differentiate between these examples. Let me know if that is fine.

**Comment by nikhitasharma** [comment]
> @nikhitasharma - great job on the documentation. I have one minor observation - in the example section of the about page BasicSlider is listed, but in the vertical navigation menu this item is called Default Slider.

Addressed this [here](https://github.com/cerner/terra-framework/pull/1945/commits/715f9208bff9a79fe767e7cc2d97decac2dff644) by making the 'Default Slider' terminology consistent.

**Comment by cm9361** [comment]
Is Page Up/Down always 10 steps?  If I have a range of 0-20, will it still be 10 step increments?

**Comment by cm9361** [comment]
I don't think Tab and Shift+Tab should be part of the keyboard interaction list.  This control is not processing them so you cannot guarantee the result.

**Comment by cm9361** [comment]
In this example, https://engineering.cerner.com/terra-framework/pull/1945/components/cerner-terra-framework-docs/slider/examples/slider-with-value-update, shouldn't your value state variable initialize to a number instead of a string?

**Comment by mjpalazzo** [comment]
@nikhitasharma - Looks good! Approved

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx (hunk 5 above)
Comment (chrismichalewicz) [reply packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx]: Can you add "the" in front of slider?

Comment (nikhitasharma) [reply packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx]: Added

**Comment by nikhitasharma** [comment]
> Is Page Up/Down always 10 steps? If I have a range of 0-20, will it still be 10 step increments?

Changed the wording as discussed offline.

**Comment by nikhitasharma** [comment]
> I don't think Tab and Shift+Tab should be part of the keyboard interaction list. This control is not processing them so you cannot guarantee the result.

Removed these.

**Comment by nikhitasharma** [comment]
> In this example, https://engineering.cerner.com/terra-framework/pull/1945/components/cerner-terra-framework-docs/slider/examples/slider-with-value-update, shouldn't your value state variable initialize to a number instead of a string?

Updated this.

**Review by chrismichalewicz — APPROVED** [review]
Looks good @nikhitasharma

**Comment by nikhitasharma** [comment]
> Fails
> 🚫	Please include a CHANGELOG entry for each changed package this PR. Looks like a CHANGELOG entry is missing for:
> * terra-slider
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [a649b4e](https://github.com/cerner/terra-framework/commit/a649b4eb5ba1bf97603d06eafefd594a450e8222)

I don't think a changelog entry is required for just property description updates.

**Review by cldeitrich — APPROVED** [review]
[OHAIUAD-205124](https://jira3.cerner.com/browse/OHAIUAD-205124): Approved for style and formatting.

File: packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx [hunk 10]
```diff
@@ -2,56 +2,101 @@ import { Notice } from "@cerner/terra-docs";
...
+* [Examples](/components/cerner-terra-framework-docs/slider/about#examples)
+* [Accessibility](/components/cerner-terra-framework-docs/slider/about#accessibility)
+* [Implementation](/components/cerner-terra-framework-docs/slider/about#implementation)
+* [Testing](/components/cerner-terra-framework-docs/slider/about#testing)
+* [Properties](/components/cerner-terra-framework-docs/slider/about#properties)
+* [Terra Standards](/components/cerner-terra-framework-docs/slider/about#terra-standards)
+
+## Overview
+<Notice variant="important" ariaLevel="2">
+The Terra Slider is an input component used to select an imprecise value within a range such as a zoom level. The component does not currently allow you to enter precise values.
```
Comment (eawww) [inline packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx:22]:
```suggestion
The Terra Slider is an input component used to select an imprecise value within a range such as a zoom level. The component does not currently support entry of precise values.
```

**Pull request merged**